frontend/src/pages/modelServing/screens/projects/nimUtils.ts

```diff
@@ -123,3 +124,60 @@ export const updateServingRuntimeTemplate = (
   }
   return updatedServingRuntime;
 };
+
+export const getNIMResourcesToDelete = async (
+  projectName: string,
+  servingRuntime: ServingRuntimeKind,
+): Promise<Promise<void>[]> => {
+  const resourcesToDelete: Promise<void>[] = [];
+  try {
+    const count = await fetchInferenceServiceCount(projectName);
+    if (count === 0) {
+      return resourcesToDelete;
+    }
+    const pvcName = servingRuntime.spec.volumes?.find((vol) =>
+      vol.persistentVolumeClaim?.claimName.startsWith('nim-pvc'),
+    )?.persistentVolumeClaim?.claimName;
+
+    let nimSecretName: string | undefined;
+    let imagePullSecretName: string | undefined;
+
+    servingRuntime.spec.containers.some((container) =>
+      container.env?.some((env) => {
```

Comment:
Probably meant to do a .forEach here -- it's odd to use a `.some` and store the data outside of it

Reply:
Yes, updated.